### PR TITLE
Re str fixes

### PR DESCRIPTION
**Pull request by rgrinberg**
Details in the commit messages

@Drup 

cc @vouillon

### CODE REVIEW

File: lib/re.mli [hunk 1]
```diff
@@ -95,7 +95,11 @@ module Group : sig
   val test : t -> int -> bool
   (** Test whether a group matched *)
 
+  val length : t -> int
```
Comment (rgrinberg) [inline lib/re.mli]: @Drup a better name here would be welcome.

**Comment by rgrinberg** [comment]
Will merge tomorrow if there are no objections.

File: lib/re.mli (hunk 1 above)
Comment (Drup) [inline lib/re.mli]: `nb_groups` ? or some other combination of "number" and "groups".

**Comment by Drup** [comment]
Seems okay to me. I'm not sure I understand the commit https://github.com/ocaml/ocaml-re/pull/110/commits/b26158a9e38f0450e55de2bcff275ceb5322c6ad, are those feature not supported at all, or just not part of the `str` interface ?

**Comment by rgrinberg** [comment]
Yes, https://github.com/ocaml/ocaml-re/pull/110/commits/b26158a9e38f0450e55de2bcff275ceb5322c6ad removes tests for features that aren't part of Str at all (they are emacs regex syntax)

**Comment by rgrinberg** [comment]
length is now nb_groups. Open to further renaming/marking this as experimental

**Pull request merged**